DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -69,23 +84,45 @@ public IActionResult EditDetails(EditDetailsViewModel model, string action)
 
         private IActionResult EditDetailsPostSave(EditDetailsViewModel model)
         {
+            ValidateCustomPrompts(model);
+
             if (!ModelState.IsValid)
             {
+                ViewBag.JobGroupOptions = GetJobGroupItems(model.JobGroupId);

```

Comment:
Feels like there is a lot of setting of ViewBag.JobGroupOptions and CustomFields as it is done each time a different problem occurs. Is there a way of commonising this, or perhaps using something similar to ViewBag so the values are retained? If it is too big an issue, don't worry about it.